qhana_plugin_registry/db/pagination.py

```diff
@@ -88,7 +88,9 @@ def get_page_info(
         sort_column = sortables[col_name]
         if "collate" in sort_column.info:
             order_by_clauses.append(
-                sort_direction(sort_column.collate(sort_column.info["collate"]))
+                sort_direction(
+                    sort_column.collate(sort_column.info["collate"][DB.engine.name])
```

Comment:
This will produce some errors that may be hard to track down if the `collate` info is not in the right format or missing an engine. Because this is inside an often used (but not often changed) utility, it may be better to catch these cases and output a [warning](https://docs.python.org/3/library/warnings.html) (warnings do not interrupt execution) or an error. If you use a warning, then fall back to not specifying the collation.

Reply:
It is now checked whether the collate is specified as dictionary and ensures that the dictionary has a key corresponding to `DB.engine.name` before the collate it is used.

https://github.com/UST-QuAntiL/qhana-plugin-registry/blob/83bb0a54c3039d965871d802867cb9cd6f037955/qhana_plugin_registry/db/pagination.py#L90-L102